torch/csrc/autograd/python_function.cpp

```diff
@@ -564,6 +564,11 @@ static void _trace_post_record(
   }
 
   node->i_(jit::attr::inplace, is_inplace);
+  if (auto module_name = PyDict_GetItemString(((PyTypeObject*)op_obj)->tp_dict, "__module__")) {
```

Comment:
@albanD does this change looks good to you?

Reply:
cc original author @shubhambhokare1. BTW @albanD reviewed the original PR I believe #67193.

Reply:
I would prefer to have a `PyObject*` instead of `auto` here tbh. The returned object is a borrowed reference so we should not handle its lifetime and the way we represent that in the codebase is by using these plain pointers.

Reply:
updated

Reply:
Thanks for the update, it looks good